aws/src/main/java/org/apache/iceberg/aws/glue/GlueTableOperations.java

```diff
@@ -117,9 +120,39 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
       if (exceptionThrown) {
         io().deleteFile(newMetadataLocation);
       }
+
+      unlock();
+    }
+  }
+
+  private void acquireLock() {
+    try {
+      boolean lockAcquired = lockManager.tryLock(lockId(databaseName, tableName),
+          awsProperties.glueCatalogLockExpireMs(), awsProperties.glueCatalogLockWaitMs());
+      if (!lockAcquired) {
+        throw new CommitFailedException("Timed out after %s ms waiting for lock on %s.%s",
+            awsProperties.glueCatalogLockWaitMs(), databaseName, tableName);
+      }
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new CommitFailedException(e, "Thread interrupted during commit");
```

Comment:
This should throw something other than `CommitFailedException` because that exception will cause a retry. When the commit is interrupted, I don't think that it should retry.